src/shape/frustum.js

```diff
@@ -1,114 +1,117 @@
-import { Mat4 } from '../math/mat4.js';
-
-var viewProj = new Mat4();
-
 /**
  * @class
  * @name pc.Frustum
- * @classdesc A frustum is a shape that defines the viewing space of a camera.
+ * @classdesc A frustum is a shape that defines the viewing space of a camera. It can be
+ * used to determine visibility of points and bounding spheres. Typically, you would not
+ * create a Frustum shape directly, but instead query {@link pc.CameraComponent#frustum}.
  * @description Creates a new frustum shape.
  * @example
- * // Create a new frustum equivalent to one held by a camera component
- * var projectionMatrix = entity.camera.projectionMatrix;
- * var viewMatrix = entity.camera.viewMatrix;
- * var frustum = new pc.Frustum(projectionMatrix, viewMatrix);
- * @param {pc.Mat4} projectionMatrix - The projection matrix describing the shape of the frustum.
- * @param {pc.Mat4} viewMatrix - The inverse of the world transformation matrix for the frustum.
+ * var frustum = new pc.Frustum();
  */
-export function Frustum(projectionMatrix, viewMatrix) {
-    projectionMatrix = projectionMatrix || new Mat4().setPerspective(90, 16 / 9, 0.1, 1000);
-    viewMatrix = viewMatrix || new Mat4();
-
+function Frustum() {
     this.planes = [];
     for (var i = 0; i < 6; i++)
         this.planes[i] = [];
-
-    this.update(projectionMatrix, viewMatrix);
 }
 
 Object.assign(Frustum.prototype, {
     /**
      * @function
-     * @name pc.Frustum#update
-     * @description Updates the frustum shape based on a view matrix and a projection matrix.
-     * @param {pc.Mat4} projectionMatrix - The projection matrix describing the shape of the frustum.
-     * @param {pc.Mat4} viewMatrix - The inverse of the world transformation matrix for the frustum.
+     * @name pc.Frustum#setFromMat4
+     * @description Updates the frustum shape based on the supplied 4x4 matrix.
+     * @param {pc.Mat4} matrix - The matrix describing the shape of the frustum.
+     * @example
+     * // Create a perspective projection matrix
+     * var projMat = pc.Mat4();
+     * projMat.setPerspective(45, 16 / 9, 1, 1000);
+     *
+     * // Create a frustum shape that is represented by the matrix
+     * var frustum = new pc.Frustum();
+     * frustum.setFromMat4(projMat);
      */
-    update: function (projectionMatrix, viewMatrix) {
-        viewProj.mul2(projectionMatrix, viewMatrix);
-        var vpm = viewProj.data;
+    setFromMat4: function (matrix) {
+        var vpm = matrix.data;
+
+        var plane;
+        var planes = this.planes;
 
         // Extract the numbers for the RIGHT plane
-        this.planes[0][0] = vpm[3] - vpm[0];
-        this.planes[0][1] = vpm[7] - vpm[4];
-        this.planes[0][2] = vpm[11] - vpm[8];
-        this.planes[0][3] = vpm[15] - vpm[12];
+        plane = planes[0];
+        plane[0] = vpm[3] - vpm[0];
+        plane[1] = vpm[7] - vpm[4];
+        plane[2] = vpm[11] - vpm[8];
+        plane[3] = vpm[15] - vpm[12];
```

Comment:
Almost repeating code - could this be made more concise with a function?

Reply:
Yep. Next PR will refactor `pc.Plane` so that `pc.Frustum` can use those instead. But I thought it would be better to break it into two steps.

Reply:
seems fine then :)